software/drivers/sd8887/src/bluetooth/bt/bt_drv.h

```diff
@@ -265,6 +268,14 @@ hexdump(char *prompt, u8 *buf, int len)
 #define os_wait_timeout(waitq, cond, timeout) \
          wait_event_timeout(waitq, cond, ((timeout) * HZ / 1000))
 
+/** timeval */
+typedef struct {
+	/** Time (seconds) */
+	t_u32 time_sec;
+	/** Time (micro seconds) */
+	t_u32 time_usec;
+} bt_timeval;
+
```

Comment:
Did you consider using the `__kernel_old_timeval` struct that was introduced in https://github.com/torvalds/linux/commit/a84d1169164b274f13b97a23ff235c000efe3b49? It seems to be a cleaner way to remain compatible while signalling that this code is not y2038 compliant.

Reply:
Right, I initially did exactly that and it was working, but later on when I struggled with the mgmt_frame_register callback I started to look around more and got lucky as I found this branch in a fork for a different chip that is maintained by NXP: https://source.codeaurora.org/external/nxpwifi/mwifiex/log/?h=caf/lf-5.10.y_1.0.0. Afterwards I reverted all my changes and used theirs instead as it will be easier to use those along the way - I see another branch now there that even support compilation on 5.12. I added this as information to the commit logs together with the kernel version supported in each commit.

Reply:
Thanks for the explanation Zahari, it's good to have an upstream reference.